### ISSUE TITLE
what is recommended restore procedure?

### ISSUE THREAD

**Issue by rlister** [markitx/dynamo-backup-to-s3#8]
Thanks for this, it works very nicely for backup.

Is there any recommended restore method for the backups produced?

**Comment by dylanlingelbach** [comment]
@rlister - the data is backed up as individual JSON items on each line. You should be able to read in the file a line at a time and then insert those items into the DynamoDB table you are restoring.

We'd be happy to take a PR adding this functionality.

Let me know if that doesn't work for you.

**Comment by GreyEcologist** [comment]
Hey @dylanlingelbach I've started looking at your script, thanks for making it available Could you elaborate on the restore procedure?

**Comment by rlister** [comment]
@GreyEcologist It's very simple, you're basically just calling `PutItem` for each line of the backup file. I'd put in a PR here but my nodejs-fu is very, very weak. Here's how I do it in ruby:

https://github.com/rlister/awful/blob/master/lib/awful/dynamodb.rb#L192-L221

You can do the same using AWS SDK for your favourite language. Note that writes are slow compared to reads (depending on indexes), so you'll likely want to crank up write_capacity if you have non-trivial data size.

**Comment by GreyEcologist** [comment]
Ok, thanks, I'll try and let you know how that worked for me.

**Comment by bradoyler** [comment]
@GreyEcologist btw, here's a node "script" that does a restore to dynamodb. https://github.com/yegor256/dynamo-archive/blob/master/bin/dynamo-restore.js

**Comment by jcbarton** [comment]
I've forked and added a new command-line argument that will export the data in a format compatible with the AWS Data Pipeline import.

This seems to require items with lowercase keys, e,g,:
```
"Count":{"n":"38"}
"ProductCategory":{"s":"Bicycle"}
"Color":{"sS":["Black","Green"]}
```
No idea why, but when I tried it using the 'default' output it failed. It seems crazy to me that it would rq

AWS Data Pipelines create an EMR cluster - so for smaller datasets it would definitely be better to use the SDK's and do ```putItem``` or ```batchWriteItem```.

If you want me to submit a pull request I will.
